grakn-bootup/src/main/java/ai/grakn/bootup/StorageConfig.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.bootup;
+
+import ai.grakn.GraknConfigKey;
+import ai.grakn.engine.GraknConfig;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author Kasper Piskorski
+ */
+public class StorageConfig {
+
+    private final ImmutableMap<String, Object> params;
+    private static final String EMPTY_VALUE = "";
+    private static final String CONFIG_PARAM_PREFIX = "storage.internal.";
+
+    private StorageConfig(Map<String, Object> p){
+        this.params = ImmutableMap.copyOf(
+                p.entrySet().stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue() == null ? EMPTY_VALUE : e.getValue()))
```

Comment:
Try to unify this operation with `mapper.readValue()` in a method called something like `parseStringtoMap`